fiftyone/operators/operations.py

```
@@ -479,12 +479,44 @@ def set_selected_samples(self, samples):
+                ``{"sample_id": "...", "type": "default"|"alt"}``
+        """
+        normalized = []
+        for item in samples:
+            if isinstance(item, str):
+                normalized.append({"sample_id": item, "type": "default"})
+            elif isinstance(item, dict):
+                normalized.append(item)
+            else:
+                raise ValueError(
+                    f"Invalid sample item: {item}. Must be a string or dict"
+                )
```

Ruff (0.15.5)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)